salt/modules/nilrt_ip.py

```diff
@@ -417,17 +417,16 @@ def _get_interface_info(interface):
     if data['adapter_mode'] == 'ethercat':
         needed_settings += ['MasterID']
     settings = _load_config(interface.name, needed_settings)
-    if interface.flags & IFF_RUNNING != 0:
-        data['up'] = True
-        data['ipv4']['address'] = interface.sockaddrToStr(interface.addr)
-        data['ipv4']['netmask'] = interface.sockaddrToStr(interface.netmask)
-        data['ipv4']['gateway'] = '0.0.0.0'
-        data['ipv4']['dns'] = _get_dns_info()
-    elif data['ipv4']['requestmode'] == 'static':
+    if data['ipv4']['requestmode'] == 'static':
         data['ipv4']['address'] = settings['IP_Address']
         data['ipv4']['netmask'] = settings['Subnet_Mask']
         data['ipv4']['gateway'] = settings['Gateway']
         data['ipv4']['dns'] = [settings['DNS_Address']]
+    elif data['up']:
```

Comment:
If this is 'up', we shouldn't _load_config at all, right?

Reply:
If the interface uses a **static** IP, then configs should be loaded using `_load_config`, no matter if the interface is up or not. If it doesn't use a **static** IP and it's up, then configs will be loaded using `pyiface` and `/etc/resolv.conf` for **DNS**

Reply:
On this 'elif data['up']' you don't use the 'settings' at all. I was asking if you can move this before doing doing the _load_config().

I would also check for 'ethercat' adapter_mode before all this tcp-related logic and handle that and return.

Reply:
That was the problem before, that I checked data['up'] first ... So, it the adapter_mode is `ethercat`, data['up'] will be `False` and data['ipv4'] may contain other information, but since we know that's ethercat, that ipv4 information is not relevant and we ignore it.

Reply:
OK, so the code looks right now, but I would refactor it to avoid confusion and the bugs that come out of it in the future, in a manner to have worker methods for 'disabled', 'ethercat', 'tcp' modes. I would have a get_base_interface_info() to return the common information for those cases to avoid duplicating the code. That way, it's going to be more maintainable in the future.